src/pages/Variants.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react';
+import { useLocation, useParams } from 'react-router-dom';
+import { VUE } from '../model/VUE';
+import { fetchVueData } from '../utils/VUEDataFetcherUtils';
+
+export const Variants: React.FC = () => {
+    const state = useLocation().state;
+    const [vue, setVue] = useState<VUE | undefined>(state);
+
+    const gene = useParams().gene;
+
+    useEffect(() => {
+        if (!state) {
+            const fetchData = async () => {
+                const data = await fetchVueData();
+                setVue(data.find(i => i.hugoGeneSymbol === gene));
+            };
+    
+            fetchData();
```

Comment:
Maybe no need to fetch data again, can we have a global store so we only need fetch once